modules/new_profile/jsx/NewProfileIndex.js

```diff
@@ -83,46 +81,75 @@ class NewProfileIndex extends React.Component {
     if (!match) {
-      this.setState({
-        isCreated: false,
-      });
-    } else {
-      let formData = this.state.formData;
-      let formObject = new FormData();
-      for (let key in formData) {
-        if (formData[key] !== '') {
-          formObject.append(key, formData[key]);
-        }
-      }
-      formObject.append('fire_away', 'New Candidate');
+      return;
+    }
+    const formData = this.state.formData;
+    const configData = this.state.configData;
 
-      // disable button to prevent form resubmission.
-      this.setState({submitDisabled: true});
+    let candidateObject = {
+        'Candidate': {
+            'Project': configData.project[formData.project],
+            // 'PSCID' : conditionally included below
+            // 'EDC' : conditionally included below
+            'DoB': formData.dobDate,
+            'Sex': formData.sex,
+            'Site': configData.site[formData.site],
+        },
+    };
+
+    if (this.state.configData['edc'] === 'true') {
+        candidateObject.Candidate.EDC = formData.edc;
+    }
+    if (this.state.configData['pscidSet'] === 'true') {
+      candidateObject.Candidate.PSCID = formData.pscid;
+    }
 
-      fetch(this.props.submitURL, {
-        method: 'POST',
-        cache: 'no-cache',
-        credentials: 'same-origin',
-        body: formObject,
-        })
-      .then((resp) => {
-        if (resp.ok && resp.status === 201) {
-          resp.json().then((data) => {
-            this.setState({newData: data});
-            this.setState({isCreated: true});
+    // disable button to prevent form resubmission.
+    this.setState({submitDisabled: true});
+
+    fetch(this.props.submitURL, {
+      method: 'POST',
+      cache: 'no-cache',
+      credentials: 'same-origin',
+      body: JSON.stringify(candidateObject),
+    })
+    .then((resp) => {
+      if (resp.ok && resp.status === 201) {
+        resp.json().then((data) => {
+            swal.fire({
+                type: 'success',
+                title: 'New Candidate Created',
+                html: 'DCCID: ' + data.CandID + ' '
+                      + 'PSCID: ' + data.PSCID + ' ',
+                confirmButtonText: 'Access Profile',
+                // Repurpose "cancel" as "recruit another candidate".
+                // Use the same colour for both buttons, since one
+                // isn't more "right" than the other.
+                showCancelButton: true,
+                cancelButtonColor: '#3085d6',
+                cancelButtonText: 'Recruit another candidate',
+            }).then((result) => {
+                if (result.value === true) {
+                    window.location.href = '/' + data.CandID;
+                } else {
+                   this.setState({
+                       formData: {},
+                       submitDisabled: false,
+                   });
+                }
+            });
           });
         } else {
           resp.json().then((message) => {
```

Comment:
I would say thta it is because the class is extending `\NDB_Form` and not `Endpoint`. It's wraped by a UserPage Decorator middleware that only prints html. Regardless of specifics, there are cases where html will be returned and `resp.json()` will not be happy. ONe way of handling that is to put the 
```
this.setState({submitDisabled: false});
swal.fire('Error!', message.error, 'error');
```
parts in the catch; and throw resp.json().message in the else.